smoke-tester/src/main.rs

```diff
@@ -372,16 +428,37 @@ impl<'a> TestTracker<'a> {
             Ok(()) => {
                 println_test_status!(self, green, "Test passed in {formatted_duration}.");
             }
+            Err(TestFailure::UnimplementedForTarget(target, message)) => {
+                println_test_status!(
+                    self,
+                    yellow,
+                    "Test not implemented for {}: {}",
+                    target.name,
+                    message
+                );
+            }
+            Err(TestFailure::MissingResource(message)) => {
+                println_test_status!(self, yellow, "Missing resource for test: {}", message);
+            }
             Err(_e) => {
                 println_test_status!(self, red, "Test failed in {formatted_duration}.");
             }
         };
         self.advance_test();
 
-        test_result
+        SingleTestReport {
+            result: test_result,
+            _duration: duration,
+        }
     }
 }
 
+type TestFn = fn(&TestTracker, &mut probe_rs::Core) -> TestResult;
+
 /// A list of all tests which run on cores.
 #[distributed_slice]
-pub static CORE_TESTS: [fn(&TestTracker, &mut probe_rs::Core) -> Result<(), probe_rs::Error>];
+pub static CORE_TESTS: [TestFn];
+
+/// A list of all tests which run on `Session`.
+#[distributed_slice]
+pub static SESSION_TESTS: [fn(&TestTracker, &mut probe_rs::Session) -> TestResult];
```

Comment:
Just a style thing, why did you extract a type for CORE_TESTS but not for SESSION_TESTS?

Reply:
No idea, I've removed the alias. Seems pointless when it's only used once.